clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2760,9 +2760,19 @@ void handleFailure(Throwable throwable) {
             }, now);
         }
 
-        return new DescribeConfigsResult(new HashMap<>(nodeFutures.entrySet().stream()
+        return new DescribeConfigsResult(
+            nodeFutures.entrySet()
+                .stream()
                 .flatMap(x -> x.getValue().entrySet().stream())
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))));
+                .collect(Collectors.toMap(
+                    Map.Entry::getKey,
+                    Map.Entry::getValue,
+                    (oldValue, newValue) -> {
+                        throw new IllegalStateException(String.format("Duplicate key for values: %s and %s", oldValue, newValue));
```

Comment:
Could you please add comments to remind readers that should not happen.

Maybe we can add test which using duplicate `resources` as input for `describeConfigs`

Reply:
Sounds good, I will do that.